app/src/main/java/com/example/practiceapp/LoginActivity.kt

```diff
@@ -23,25 +23,24 @@ class LoginActivity : AppCompatActivity() {
         }
 
         binding.loginBtnLogin.setOnClickListener {
-
             val emaill = binding.loginEtUser.text.toString()
             val pass = binding.loginEtpass.text.toString()
-            val db = (application as UserApp).database
-            val userdb = db.userDao().getById(emaill)
-            val emaildb = userdb.email
-            val passdb = userdb.password
-            if(emaildb.isBlank()){
-                if (emaill == emaildb && passdb == pass){
-                    startActivity(Intent(this, BuscarActivity::class.java))
-                }else{
-                    Toast.makeText(this, " Usuario o contraseña incorrecto", Toast.LENGTH_SHORT).show()
+            if (emaill.isEmpty() || pass.isEmpty()) {
+                Toast.makeText(this, "Debes ingresar ambos campos para iniciar sesion", Toast.LENGTH_SHORT).show()
+            } else {
+                val db = (application as UserApp).database
+                val userdb = db.userDao().getById(emaill)
+                if (userdb != null) {
+                    val passdb = userdb.password
+                    if (passdb == pass) {
```

Comment:
you can avoid this validation by having a method and query in the DAO that searches by email and password instead of only searching by email.